src/palette/qml/MuseScore/Palette/internal/PaletteTree.qml

```diff
@@ -114,6 +114,9 @@ StyledListView {
             // in search and no cell selected: apply the first found element
             const parentIndex = paletteTreeDelegateModel.modelIndex(0);
             index = paletteModel.index(0, 0, parentIndex);
+        } else if (!index.valid) {
+            // no cell selected and not in search: use current index from previous click
+            index = currentIndex;
```

Comment:
This seems incorrect to me. `paletteController.applyPaletteElement` expects a QModelIndex, not an int, like `currentIndex` is. 

Also, when I run it locally, this code is never called. For me, it works fine without these three added lines, regardless of whether the current palette element was selected using keyboard navigation or by clicking on it.

Reply:
Yes, I figured the type mismatch when I tried running it from the build artifact.  Like I said, I'm really just guessing here!  For me, it definitely doesn't work, though.  The code above *is* called - I got error messages to the console about the mismatch (which is how I discovered it).  Strange that it's otherwise for you, though.  Anyhow, trying something else now...